src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1214,6 +1213,24 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
                 base.VisitMarkupElement(node);
 
                 _builder.Pop();
+
+                static bool LooksLikeAComponentName(DocumentIntermediateNode document, string startTagName)
+                {
+                    var category = char.GetUnicodeCategory(startTagName, 0);
+
+                    // A markup element which starts with an uppercase character is likely a component.
+                    //
+                    // In certain cultures, characters are not explicitly Uppercase/Lowercase, hence we must check
+                    // the specific UnicodeCategory to see if we may still be able to treat it as a component.
+                    //
+                    // The goal here is to avoid clashing with any future standard-HTML elements.
+                    //
+                    // To avoid a breaking change, the support of localized component names (without explicit
+                    // Uppercase classification) is behind a `SupportLocalizedComponentNames` feature flag.
+                    return category is UnicodeCategory.UppercaseLetter ||
+                        (document.Options.SupportLocalizedComponentNames &&
+                            (category is UnicodeCategory.TitlecaseLetter || category is UnicodeCategory.OtherLetter));
```

Comment:
Are these the only categories?

Reply:
Yes, these are the only ones we'd like to support for Blazor component names based on discussions with @tarekgh.

Reply:
These are the category we need to match validating the component name which is needed to start with Uppercase letter (or a letter with no casing).